specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/Trigger.json

```diff
@@ -365,6 +372,86 @@
           "maximum": 86400
         }
       }
+    },
+    "TriggerReference": {
+      "description": "Trigger reference type.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Trigger reference type.",
+          "type": "string",
+          "enum": [
+            "TriggerReference"
+          ]
+        },
+        "referenceName": {
+          "description": "Reference trigger name.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "type",
+        "referenceName"
+      ]
+    },
+    "DependencyReference": {
+      "description": "Referenced dependency",
+      "type": "object",
+      "discriminator": "type",
+      "properties": {
+        "type": {
+          "description": "The type of dependency reference",
+          "type": "string"
+        }
+      },
+      "required": [
+        "type"
+      ]
+    },
+    "TumblingWindowTriggerDependencyReference": {
+      "description": "Referenced tumbling window trigger dependency",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DependencyReference"
+        }
+      ],
+      "properties": {
+        "referenceTrigger": {
+          "description": "Referenced trigger",
+          "$ref": "#/definitions/TriggerReference"
+        },
+        "offset": {

```

Comment:
offset and size can be moved to a separate type and reused in both self dependency and tumbling widow dependency reference classes. Please refer to the mail thread and the doc Nailini created.

Reply:
Also it will be better to create another class derived from DependencyReference as Nalini had in her doc and put referenceTrigger in that class, because in future if you support ScheduledTriggerReference, it will be derived from this new class along with TumblingWindowTriggerDependencyReference. Only in case of self rereference you don't need to have referenceTrigger property, in all other cases you will need it. If we don't do these changes now, later they will breaking changes.

Reply:
Offset is required for self-dependency, so it makes not much sense to create a separate type for offset and size.

Reply:
TriggerDependencyReference has been added